core/src/test/java/io/grpc/internal/FakeClock.java

```diff
@@ -50,8 +51,8 @@ public boolean shouldAccept(Runnable command) {
 
   private final ScheduledExecutorService scheduledExecutorService = new ScheduledExecutorImpl();
 
-  private final PriorityBlockingQueue<ScheduledTask> tasks =
-      new PriorityBlockingQueue<ScheduledTask>();
+  // Must keep the ordering of tasks as they are required by ControlPlaneScheduler.scheduleNow().
+  private final LinkedBlockingQueue<ScheduledTask> tasks = new LinkedBlockingQueue<>();
```

Comment:
Can we simply use two queues instead? One for pending (ready to be executed) tasks and one for scheduled (for a future time) tasks? We'd keep the previous PriorityBlockingQueue and then just add a LinkedBlockingQueue for execute(). That more closely matches what would happen in practice and makes the code more clear.

Reply:
Done.